h/static/scripts/directive/annotation.js

```diff
@@ -0,0 +1,730 @@
+      $scope.$digest();
+    }, nextUpdate, false);
+  };
+
+  // Export the baseURI for the share link.
+  vm.baseURI = $document.prop('baseURI');
+
+  $scope.$on('$destroy', function() {
+    updateTimestamp = angular.noop;
+  });
+
+  // Watch for changes to the domain model and recreate the view model when it
+  // changes.
+  // XXX: TODO: don't clobber the view when collaborating.
+  $scope.$watch((function() {return model;}), function(model, old) {
+    if (model.updated !== old.updated) {
+      // Discard saved drafts.
+      drafts.remove(model);
+    }
+
+    // Save highlights once logged in.
+    if (vm.isHighlight() && highlight) {
+      if (model.user && !model.id) {
+        model.permissions = permissions.private();
+        model.$create().then(function() {
+          $rootScope.$emit('annotationCreated', model);
+        });
+        highlight = false;  // Prevents double highlight creation.
+      } else {
+        updateDraft(model);
+      }
+    }
+
+    updateTimestamp(model === old);  // Repeat on first run.
+    vm.render();
+  }, true);
+
+  $scope.$on(events.USER_CHANGED, function() {
+    if (model.user === null) {
+      model.user = session.state.userid;
+    }
+
+    // Set model.permissions on sign in, if it isn't already set.
+    // This is because you can create annotations when signed out and they
+    // will have model.permissions = null, then when you sign in we set the
+    // permissions correctly here.
+    if (!model.permissions) {
+      model.permissions = permissions['default'](model.group);
+    }
+  });
+
+  // If this is a new annotation or we have unsaved changes,
+  // then start editing immediately.
+  isNewAnnotation = !(model.id || (vm.isHighlight() && highlight));
+  if (isNewAnnotation || drafts.get(model)) {
+    vm.edit();
+  }
+
+  // When the current group changes, persist any unsaved changes using
+  // the drafts service. They will be restored when this annotation is
+  // next loaded.
+  $scope.$on(events.GROUP_FOCUSED, function() {
+    var draftDomainModel;
+    var isShared;
+    var newGroup;
+
+    if (!vm.editing) {
+      return;
+    }
+
+    // Move any new annotations to the currently focused group when
+    // switching groups. See GH #2689 for context.
+    if (!model.id) {
+      newGroup = groups.focused().id;
+      isShared = permissions.isShared(
+        vm.annotation.permissions, vm.annotation.group);
+      if (isShared) {
+        model.permissions = permissions.shared(newGroup);
+        vm.annotation.permissions = model.permissions;
+      }
+      model.group = newGroup;
+      vm.annotation.group = model.group;
+    }
+
+    // if we have a draft, update it, otherwise (eg. when the user signs out)
+    // do not create a new one.
+    if (drafts.get(model)) {
+      draftDomainModel = {};
+      updateDomainModel(draftDomainModel, vm.annotation);
+      updateDraft(draftDomainModel);
+    }
+  });
+
+  return vm;
+}
+
+// @ngInject
```

Comment:
Does `// @ngInject` work if it isn't directly above the function expression? Even if it does, it would be more consistent to put it just above the function or method declaration.